README.md

```diff
@@ -1,2 +1,243 @@
-# intro-to-git
-An group exercise practicing with various git ideas and tools
+# Introduction to `git`
+
+This will be an in-lab exercise where we'll experience several of the
+key features of the `git` version control system, and the Github
+repository hosting service.
+
+:bangbang: While in most labs it will be fine for groups to move ahead
+at their own pace, in this lab we'd like people to keep together
+because there are moments where we're expecting certain things to
+happen (e.g., merge conflicts) and we want everyone to be in roughly
+the same place when that happens.
+
+This definitely will _not_ provide a comprehansive overview of the
+_many_ features `git` provides. There are a lot of on-line resources
+that can provide additional information, including:
+
+* The excellent [Atlassian `git` tutorials](https://www.atlassian.com/git/tutorials/what-is-version-control)
+* A very nice [on-line tutorial from Code School](https://try.github.io/levels/1/challenges/1)
+* [The "standard" `git` documentation site](https://git-scm.com/documentation),
+which also includes links to videos, cheat sheets, and such
+* [`git` – the simple guide](http://rogerdudler.github.io/git-guide/),
+a single-page app that goes through the major features of `git`
+* [The original blog post on the Gitflow branching model](http://nvie.com/posts/a-successful-git-branching-model/)
+(and [a later post arguing that GitFlow is a Very Bad Idea](http://endoflineblog.com/gitflow-considered-harmful))
+* [Git Essential Training](https://www.lynda.com/Git-tutorials/Git-Essential-Training/100222-2.html),
+an on-line Lynda.com course; it's free if you first authenticate
+with your U of M credentials via http://lynda.umn.edu
+
+The discussion below assumes that people are paired up in the lab, and
+there will be times where one person will need to log out and let their
+partner login so that everyone has a chance to go through all the
+activities.
+
```

Comment:
That would work on for the web, but it won't work for editing in WebStorm locally, committing, pushing, etc.

There are ways to hack around that, but they're ugly and error prone, so having people log out/in may be the safest/simplest option.